rotkehlchen/externalapis/helius.py

```diff
@@ -116,9 +110,19 @@ def _query(
 
         return json_ret
 
+    def maybe_warn_missing_key(self) -> None:
+        """Warns the user once if the Helius api key is missing."""
+        if not self.warning_given:
+            self.db.msg_aggregator.add_message(
+                message_type=WSMessageType.MISSING_API_KEY,
+                data={'service': ExternalService.HELIUS.serialize()},
+            )
+            self.warning_given = True
+
     def maybe_get_rpc_node(self) -> WeightedNode | None:
         """Returns the Helius RPC node if the user has an api key."""
         if (api_key := self._get_api_key()) is None:
```

Comment:
why not call this inside `_get_api_key()`? everywhere you call `self.maybe_warn_missing_key()`, you've already called `_get_api_key()` first anyway.